src/middlewared/middlewared/plugins/netdata.py

```diff
@@ -0,0 +1,241 @@
+            with open(path + file, 'r') as f:
+                for alarm in re.findall(pattern, f.read()):
+                    alarms[alarm.strip()] = path + file
+
+        return alarms
+
+    @private
+    async def validate_attrs(self, data):
+        verrors = ValidationErrors()
+
+        additional_params = data.get('additional_params')
+        if additional_params:
+            # Let's be very generic here and introduce very basic validation
+            # Expected format is as following
+            # [ipv6.icmpneighbor]
+            #   history = 86400
+            #   enabled = yes
+            #
+            # While we are here, we will also introduce basic formatting to the file to ensure
+            # that we can make it as compliable as possible
+
+            param_str = ''
+            for i in additional_params.split('\n'):
+                i = i.strip()
+                if not i:
+                    continue
+                if i.startswith('#'):
+                    # Let's not validate this
+                    if i.replace('#', '').startswith('['):
+                        param_str += f'\n\n{i}'
+                    else:
+                        param_str += f'\n\t{i}'
+
+                    continue
+
+                if i.startswith('[') and not i.endswith(']'):
+                    verrors.add(
+                        'netdata_update.additional_params',
+                        f'Please correct format for {i}. i.e [system.intr]'
+                    )
+                elif not i.startswith('[') and '=' not in i:
+                    verrors.add(
+                        'netdata_update.additional_params',
+                        f'Please correct format for {i}. i.e enabled = yes'
+                    )
+
+                if i.startswith('['):
+                    param_str += f'\n\n{i}'
+                else:
+                    param_str += f'\n\t{i}'
+
+            data['additional_params'] = param_str + '\n'
+        else:
+            # Let's load up the default value for additional params
+            # This is sort of a rollback to default configuration if a blank string is provided for
+            # additional params - we will default to the defaults of netdata.conf giving users a chance
+            # to come back to default configuration if they messed something up pretty bad
+            with open('/usr/local/etc/netdata/netdata.conf.sample', 'r') as file:
+                try:
+                    data['additional_params'] = file.read().split('per plugin configuration')[1]
+                except IndexError:
+                    self.logger.debug('Failed to set default value for additional params')
+
+        bind_to_ips = data.get('bind')
+        if bind_to_ips:
+            valid_ips = [ip['address'] for ip in await self.middleware.call('interfaces.ip_in_use')]
+            valid_ips.extend(['127.0.0.1', '::1', '0.0.0.0', '*'])
```

Comment:
`*` is not an IP, I dont think that should be valid.

Reply:
Netdata accepts it in this case and what most users would be familiar with to use here

Reply:
The API does not have to reflect the taste of netdata, lets remove * please.

Reply:
Perfect. Will have it updated : )